### PR TITLE
[LARAVEL 8] include_factory_builders does not work anymore (#1047)

### PR DESCRIPTION
**Pull request by finagin**
## Summary

See #1047 thread

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Misc. change (internal, infrastructure, maintenance, etc.)

### Checklist
- [x] Existing tests have been adapted and/or new tests have been added
- [ ] Add a CHANGELOG.md entry
- [x] Update the README.md
- [x] Code style has been fixed via `composer fix-style`

### CODE REVIEW

File: src/Factories.php [hunk 1]
```diff
@@ -26,4 +30,9 @@ public static function all()
 
         return $factories;
     }
+
+    protected static function isLaravel8(): bool
+    {
+        return version_compare(app()->version(), '8.0.0', '>=');
```
[the remark this answers is not in the document]
Comment (mfn) [reply src/Factories.php:36]: Nevermind, it's safe because it's on the `\Illuminate\Contracts\Foundation\Application` contract.

Comment (mfn) [inline src/Factories.php:36]: However 😅 the question still stands regarding Lumen:
- Lumen returns something like `Lumen (8.0.0) (Laravel Components ^8.0)`
- Laravel returns `8.0.3

`version_compare` won't see Lumen >= 8 as `true`. But it shouldn't, no idea 🤷‍♀️

**Review by mfn — COMMENTED** [review]
I think we also need to adapt the config and readme so it's clear this is obsolete/no-op and will be removed in the future (assuming we're all in the clear this isn't necessary due to the nature of change Laravel did here)

File: src/Factories.php (hunk 1 above)
Comment (xtrime-ru) [reply src/Factories.php:36]: Im using this hotfix: 

```
if (class_exists('Illuminate\Database\Eloquent\Factory')) {
//do legacy stuff...
}
```

Comment (mfn) [reply src/Factories.php:36]: Yeah, maybe that's better / more clear.

I don't use this part so I've a hard time giving useful feedback there 😬

**Review by faytekin — APPROVED** [review]

**Comment by mfn** [comment]
@finagin did you see my last comment https://github.com/barryvdh/laravel-ide-helper/pull/1049#pullrequestreview-486102621 ?

**Comment by mfn** [comment]
Thanks, I'll try to check it over the weekend!

File: README.md [hunk 2]
```diff
@@ -256,6 +256,7 @@ Then run `php artisan ide-helper:generate`, you will now see all Fluent methods
 
 If you would like the `factory()->create()` and `factory()->make()` methods to return the correct model class,
 you can enable custom factory builders with the `include_factory_builders` line your `config/ide-helper.php` file.
+Deprecated for Laravel 8 or latest.
```
Comment (mfn) [inline README.md:259]: Maybe it's my English, but shouldn't this be (same for the one in the config)
```suggestion
Deprecated for Laravel 8 or later.
```

**Comment by mfn** [comment]
Closing / Opening PR to retrigger the tests; seems some of them aborted for no apparent reason (to us)

**Review by mfn — APPROVED** [review]
I guess this is good 🤞 I don't use this feature, so hard to judge

@barryvdh let's give it a try and merge?

**Pull request merged**